lib/Dialect/Secret/IR/SecretOps.cpp

```diff
@@ -20,6 +20,24 @@ ParseResult YieldOp::parse(OpAsmParser &parser, OperationState &result) {
                  parser.resolveOperands(opInfo, types, loc, result.operands));
 }
 
+LogicalResult YieldOp::verify() {
+  // Trait verifier ensures parent is a GenericOp
+  auto parent = llvm::cast<GenericOp>(getParentOp());
+
+  for (int i = 0; i < getValues().size(); ++i) {
+    auto yieldSecretType = SecretType::get(getValues().getTypes()[i]);
+    if (yieldSecretType != parent.getResultTypes()[i]) {
+      return emitOpError()
+             << "Result types of yield op must match result types of enclosing "
+                "generic op. Yield op result types: "
+             << getValues().getTypes()
```

Comment:
should this be yieldSecretType ?

(Im not entirely familiar with how MLIR error messages work, but I would assume it will throw some form of array indexing error)

Reply:
Clarified